lib/browser/tag/tag.js

```diff
@@ -171,10 +173,7 @@ function Tag(impl, conf, innerHTML) {
     // parse layout after init. fn may calculate args for nested custom tags
     parseExpressions(dom, self, expressions)
 
-    // unmount automatically when our parent does
-    if (parent) parent.on('unmount', self.unmount)
```

Comment:
Where is this handled now?  It seems to have created a problem with dynamic `data-is` tags now go out of sync in the `parent.tags`.  It seems to partially work and all test pass if I add [here](https://github.com/riot/riot/blob/next/lib/browser/tag/update.js#L196) `parent.on('unmount', function() { expr.tag.unmount()})`.  

You can see the issue [here](http://plnkr.co/edit/K3OPjg?p=preview), just hit the add button, then remove what you added and look at the console, you will see `{string: Array[2]}` but it should return to `{string: Tag}`.  The above takes care of part of the problem for simple removals.  

There is another issue when a tag is swapped to a different type.  The code [here](https://github.com/riot/riot/blob/next/lib/browser/tag/update.js#L182) used to take care of that, but now this code does not get reached.  Any insight on how this can be handled would be appreciated.